identity-account/src/account/account.rs

```diff
@@ -247,54 +282,66 @@ impl Account {
     &self,
     old_state: &IdentityState,
     new_state: &IdentityState,
+    fragment: &Option<String>,
     document: &mut IotaDocument,
   ) -> Result<()> {
-    if self.chain_state().is_new_identity() {
-      let method: &IotaVerificationMethod = new_state.document().default_signing_method()?;
-      let location: KeyLocation = new_state.method_location(
-        method.key_type(),
-        // TODO: Should be a fatal error.
-        method.id().fragment().ok_or(Error::MethodMissingFragment)?.to_owned(),
-      )?;
-
-      // Sign the DID Document with the current capability invocation method
-      new_state
-        .sign_data(self.did(), self.storage(), &location, document)
-        .await?;
+    let (signing_state, signing_document): (&IdentityState, &IotaDocument) = if self.chain_state().is_new_identity() {
+      (new_state, new_state.document())
     } else {
-      let method: &IotaVerificationMethod = old_state.document().default_signing_method()?;
-      let location: KeyLocation = new_state.method_location(
-        method.key_type(),
-        // TODO: Should be a fatal error.
-        method.id().fragment().ok_or(Error::MethodMissingFragment)?.to_owned(),
-      )?;
-
-      // Sign the DID Document with the previous capability invocation method
-      old_state
-        .sign_data(self.did(), self.storage(), &location, document)
-        .await?;
-    }
+      (old_state, old_state.document())
+    };
+
+    let signing_method: &IotaVerificationMethod = match fragment {
+      Some(fragment) => signing_document.resolve_signing_method(fragment)?,
+      None => signing_document.default_signing_method()?,
+    };
```

Comment:
Opinion: using just the `signing_state` might be cleaner? Not sure, completely optional.

E.g.
```rust
    let signing_state: &IdentityState = if self.chain_state().is_new_identity() {
      new_state
    } else {
      old_state
    };

    let signing_method: &IotaVerificationMethod = match fragment {
      Some(fragment) => signing_state.document().resolve_signing_method(fragment)?,
      None => signing_state.document().default_signing_method()?,
    };
```